packages/business/src/api/mentorships.js

```diff
@@ -2,18 +2,45 @@ const express = require('express')
 const Router = express.Router
 const router = Router()
 const Mentorship = require('../models/Mentorship')
+const User = require('../models/User')
+const Agent = require('../models/Agent')
+
+const permission = require('../permissions')
+const permissions = permission.permissions
+const hasPermission = permission.hasPermission
 
 router.get('/', async (req, res, next) => {
-  try {
-    const docs = await Mentorship.find()
-      .populate('agents')
-      .populate('technologies')
-      .populate('mentorshipLead')
-    res.status(200).json({ docs })
-  } catch (err) {
-    console.error('An error occurred:', err)
-    next(err)
-  }
+  User.findOne({ googleId: req.user.googleId }, 'role _id').then(async userDoc => {
```

Comment:
Consider refactoring all DB lookups to async/await. This leaves the code to read as a flat level rather than nesting.